content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -1,350 +1,482 @@
 
 ```java
+
 public void setName(String name){
-    this.name = name;
+
+   this.name = name;
+
 }
 
 public String getName(){
-	return name;
+
+   return name;
+
 }
-```
 
+```
 
-So because these methods are declared with a public access modifier, other classes can set and get names whenever an object is created from the human class. So encapsulation adds security to our code by reducing how clients access our data or class contract.
+As these methods are public other classes can set and get names whenever an object is created from the human class. Encapsulation secures our code by reducing how clients access our data or class contract.
 
 In general, these two steps can be used to implement Encapsulation:
 
-- [1]: Declare class attributes as private as we did in the code above
-- [2]: Provide public setter and getters to access or manipulate instance variables, remember the setName() and getName method?
+-  Declare class attributes as private as we did in the code above
 
-###**Difference between Abstraction and Encapsulation**
+-  Provide public setter and getter methods. To access or manipulate instance variables, remember the setName() and getName method?
+
+### Difference between Abstraction and Encapsulation
 
 The obvious differences between abstraction and encapsulation are:
+
 Abstraction hides the internal details of a program and shows only the functionality
+
 Encapsulation wraps both code and data together
-Abstraction hides complexity while encapsulation hide the internal working
 
-#**Conclusion**
+Abstraction hides complexity while encapsulation hides the internal working
 
-To this point, we’ve demystified the concept of OOP starting with Classes, Objects, Abstraction, and Encapsulation and how they are implemented in Java. To say the least, abstraction and encapsulation are two sides of the same coin. I hope you find this insightful. I hope to write about Inheritance and Polymorphism in my next OOP concept article.
+# Conclusion
 
+To this point, we’ve demystified the concept of OOP. To say the least, abstraction and encapsulation are two sides of the same coin. I hope you find this insightful. I hope to write about Inheritance and Polymorphism in my next OOP concept article.
 
 ### References
+
 Java- How to program 10th Ed-Early Object Version- Deitel
-Introduction to Java Programming(10th ed) Comprehensive Version [Liang- 2014-01-06]
 
+Introduction to Java Programming (10th ed) Comprehensive Version [Liang- 2014-01-06]
+
+[oracle java docs](https://docs.oracle.com/javase/tutorial/java/IandI/abstract.html)
 
-![](https://docs.oracle.com/javase/tutorial/java/IandI/abstract.html)
-![](https://docs.oracle.com/javase/tutorial/java/concepts/index.html)
-![]([https://www.w3schools.com/java/java_abstract.asp)
-![](https://www.w3schools.com/java/java_encapsulation.asp)
+[oracle java docs](https://docs.oracle.com/javase/tutorial/java/concepts/index.html)
 
+[w3schools]([https://www.w3schools.com/java/java_abstract.asp)
```

Comment:
> [w3schools]([https://www.w3schools.com/java/java_abstract.asp)
> 
> [w3schools](https://www.w3schools.com/java/java_encapsulation.asp)

Same issue with the links to the oracle java docs. You can't distinguish what these links are for or from each other; they both look the same because they each just say: *w3schools*.